opentracing/span.lua

```diff
@@ -28,6 +29,17 @@ function span_methods:context()
   return opentracing_span_context.new()
 end
 
+-- Provides access to the @class `Tracer` that created this span.
+--
+-- @return the @class `Tracer` that created this span.
+function span_methods:tracer()
+  return self.tracer_
+end
```

Comment:
Won't this create a function that expects `span_methods` as the argument? I believe you want something like this:

```
span_mt = { ... }
span_mt.__index = span_mt

function span_mt:tracer()
  return self.tracer_
end
```

But overall, don't trust my memory of how lua metatables work. I think the whole metatable business is a bit overcomplicated. In the past, I just used [middleclass](https://github.com/kikito/middleclass) to solve these issues.

Reply:
`:` only adds an `self` argument. There's no additional runtime type checking or requirements on what `self` has to be. See https://en.wikibooks.org/wiki/Lua_Programming/colon

Reply:
OK, I see what you are doing. Instead of setting `__index` to the metatable itself, you are just dispatching to another table called `span_methods`. That's fine with me.